docs/fundamentals/gateway/custom-gateway.md

```diff
@@ -0,0 +1,326 @@
+(custom-gateway)=
+# Customize the Gateway
+A Jina Gateway is customizable and can be implemented in a similar way Executors are implemented.
+With Custom Gateways, Jina gives power back to the users, by allowing to implement any server, protocol and interface on
+the gateway level. This means you have more freedom in:
+* Defining and exposing your own API Gateway interface to clients. You can define your JSON schema or protos,...
+* Choosing your favourite server framework.
+* Choosing the protocol used to serve your app.
+
+Customization is allowed different components:
+* Implementing the custom gateway using a `base` gateway class: {class}`~jina.Gateway` or {class}`~jina.serve.runtimes.gateway.http.fastapi.FastAPIBaseGateway`.
+* Using the {class}`~jina.serve.streamer.GatewayStreamer` to send data to Executors in the Flow.
+* Implementing the needed health-checks for jina.
+* Optionally, define a `config.yml` for it.
+* Optionally, bootstrapping the gateway using a `Dockerfile` and re-using the docker image.
+
+## Implementing the custom gateway
+Similarly to how you would implement an Executor, you can implement a custom gateway by inheriting from a base gateway class.
```

Comment:
"Just like for Executors, you can implement a custom..."

Reply:
"_a_ base class" Do you mean _the_ base class?

Reply:
there are many base classes provided